### DISCUSSION TITLE
Quick Info tooltips issue

### DISCUSSION THREAD

**Discussion by idlegoose** [microsoft/vscode-cpptools#9576] [Q&A]
When mousing over the first two methods in "main" the quick info pops up as usual and show the preceding line comment from the header for the method. However, when mousing over the third line it just shows the declaration line. The majority of the quick infos do not show up for the entire source (though worked fine when I checked out last week), I just selected these three since they are declared and used in the same proximity. I have tried reverting to older versions of cpptools to no avail, and even tried the pre-release version. I could not find any quick info settings. How can I restart the Quick Info engine? I have also tried resetting the intellisense db and restarting it.

Object.h

```
// Sets the display device.
void SetDisplay(IDisplayDevice* display);

// Sets the Flow Sensor.
void SetSensor(Sensor* flow);

// Sets the Input device.
void SetInput(IInputDevice* input);
```
main.cpp
```
object->SetSensor(new Sensor());
   
object->SetDisplay(new LCD_800x600());

object->SetInput(new OneButton());
```

Any help figuring this out would be much appreciated.
-IdleGoose

**Comment by sean-mcmanus** [comment]
I'm not able to repro the issue with 

Object.h
```cpp
struct IDisplayDevice{};
struct Sensor{};
struct IInputDevice{};
struct LCD_800x600 : public IDisplayDevice{};
struct OneButton: public IInputDevice {};

class Object
{
public:
// Sets the display device.
void SetDisplay(IDisplayDevice* display);

// Sets the Flow Sensor.
void SetSensor(Sensor* flow);

// Sets the Input device.
void SetInput(IInputDevice* input);
};
```

main.cpp
```cpp
#include "Object.h"

int main()
{
    Object *object;
    object->SetSensor(new Sensor());
    
    object->SetDisplay(new LCD_800x600());

    object->SetInput(new OneButton());
}
```

Are you able to repro the issue with those files or can you provide a repro?

One thing you could try is to set C_Cpp.intelliSenseCacheSize to 0 in case it's a bug with the IntelliSense caching.

**Reply by sean-mcmanus** [reply]
![image](https://user-images.githubusercontent.com/19859882/179081880-a1f81bb9-69e8-4c21-b3bb-a5ada8f566f2.png)

Also, your OS and compiler might matter.

**Reply by idlegoose** [reply]
Forgive the late reply, I tried and have yet to reproduce the issue with any certainty.  This issue seemed to happen most often after having multiple VScode instances open at the same time for longer than a couple of hours.  

Actions that helped: Resetting IntelliSense Db for all instances-> close all VScode instances ->reboot PC -> open VScode instances one at a time and again Reset IntelliSense Db -> Rebuild IntelliSense for active file.

I have not tried setting C_Cpp.intelliSenseCacheSize to 0 as the issue was partially resolved before I read your reply.

This issue only happened about 1-2 times per month, and has not happened since the last time I reset it shortly after I posted the original message.

Should the issue resurface I will try the CacheSize setting and give you feedback.

Thank you!
-IdleGoose
